### ISSUE TITLE
Cannot open product page when options are set

### ISSUE THREAD

**Issue by gdecorbiac** [Sylius/Sylius#6230]
When I try to open the page of a product configured with options, I get a FatalThrowableError :

`Type error: Argument 1 passed to Sylius\Component\Core\Model\OrderItem::setVariant() must implement interface Sylius\Component\Core\Model\ProductVariantInterface, null given, called in [dev path]/sylius-standard/vendor/sylius/sylius/src/Sylius/Component/Core/Factory/CartItemFactory.php on line 57`

Steps to reproduce it :

- Create a product option with somes values (in my case : size options with XL, L, M, S values)
- Create a configurable product and select the newly created option
- Go to webshop and open the newly created product page

here is the full stack trace :

```
[1] Symfony\Component\Debug\Exception\FatalThrowableError: Type error: Argument 1 passed to Sylius\Component\Core\Model\OrderItem::setVariant() must implement interface Sylius\Component\Core\Model\ProductVariantInterface, null given, called in [dev path]/sylius-standard/vendor/sylius/sylius/src/Sylius/Component/Core/Factory/CartItemFactory.php on line 57
    at n/a
        in [dev path]/sylius-standard/vendor/sylius/sylius/src/Sylius/Component/Core/Model/OrderItem.php line 46

    at Sylius\Component\Core\Model\OrderItem->setVariant(null)
        in [dev path]/sylius-standard/vendor/sylius/sylius/src/Sylius/Component/Core/Factory/CartItemFactory.php line 57

    at Sylius\Component\Core\Factory\CartItemFactory->createForProduct(object(Product))
        in  line 

    at call_user_func_array(array(object(CartItemFactory), 'createForProduct'), array(object(Product)))
        in [dev path]/sylius-standard/vendor/sylius/sylius/src/Sylius/Bundle/ResourceBundle/Controller/NewResourceFactory.php line 33

    at Sylius\Bundle\ResourceBundle\Controller\NewResourceFactory->create(object(RequestConfiguration), object(CartItemFactory))
        in [dev path]/sylius-standard/vendor/sylius/sylius/src/Sylius/Bundle/OrderBundle/Controller/OrderItemController.php line 66

    at Sylius\Bundle\OrderBundle\Controller\OrderItemController->addAction(object(Request))
        in  line 

    at call_user_func_array(array(object(OrderItemController), 'addAction'), array(object(Request)))
        in [dev path]/sylius-standard/app/bootstrap.php.cache line 3238

    at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), '2')
        in [dev path]/sylius-standard/app/bootstrap.php.cache line 3197

    at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), '2', false)
        in [dev path]/sylius-standard/app/bootstrap.php.cache line 3351

    at Symfony\Component\HttpKernel\DependencyInjection\ContainerAwareHttpKernel->handle(object(Request), '2', false)
        in [dev path]/sylius-standard/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Fragment/InlineFragmentRenderer.php line 85

    at Symfony\Component\HttpKernel\Fragment\InlineFragmentRenderer->render('http://127.0.0.1:8000/cart/add?template=%40SyliusShop/Product/Show/_addToCart.html.twig&productId=1', object(Request), array('ignore_errors' => false))
        in [dev path]/sylius-standard/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Fragment/FragmentHandler.php line 136

    at Symfony\Component\HttpKernel\Fragment\FragmentHandler->render('http://127.0.0.1:8000/cart/add?template=%40SyliusShop/Product/Show/_addToCart.html.twig&productId=1', 'inline', array('ignore_errors' => false))
        in [dev path]/sylius-standard/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/DependencyInjection/LazyLoadingFragmentHandler.php line 76

    at Symfony\Component\HttpKernel\DependencyInjection\LazyLoadingFragmentHandler->render('http://127.0.0.1:8000/cart/add?template=%40SyliusShop/Product/Show/_addToCart.html.twig&productId=1', 'inline', array())
        in [dev path]/sylius-standard/vendor/symfony/symfony/src/Symfony/Bridge/Twig/Extension/HttpKernelExtension.php line 60

    at Symfony\Bridge\Twig\Extension\HttpKernelExtension->renderFragment('http://127.0.0.1:8000/cart/add?template=%40SyliusShop/Product/Show/_addToCart.html.twig&productId=1')
        in [dev path]/sylius-standard/app/cache/dev/twig/7d/7d663d87bb84b2eb8842ea8d8a14f395eb9f6d1c98fe1c383d99a187c27982d3.php line 29

    at __TwigTemplate_7367455f34ae7dca3d9afbe2a2587e8ca0c457c3940f0d46cc24aeb026756326->doDisplay(array('configuration' => object(RequestConfiguration), 'metadata' => object(Metadata), 'resource' => object(Product), 'product' => object(Product), 'app' => object(AppVariable), 'shopperContext' => object(SyliusComponentCoreContextShopperContext_000000007e497c57000000005871744d9ef656426ffaa21a64d67806c9aa9446), 'sonata_block' => object(GlobalVariables)), array())
        in [dev path]/sylius-standard/vendor/twig/twig/lib/Twig/Template.php line 379

    at Twig_Template->displayWithErrorHandling(array('configuration' => object(RequestConfiguration), 'metadata' => object(Metadata), 'resource' => object(Product), 'product' => object(Product), 'app' => object(AppVariable), 'shopperContext' => object(SyliusComponentCoreContextShopperContext_000000007e497c57000000005871744d9ef656426ffaa21a64d67806c9aa9446), 'sonata_block' => object(GlobalVariables)), array())
        in [dev path]/sylius-standard/vendor/twig/twig/lib/Twig/Template.php line 347

    at Twig_Template->display(array('configuration' => object(RequestConfiguration), 'metadata' => object(Metadata), 'resource' => object(Product), 'product' => object(Product), 'app' => object(AppVariable), 'shopperContext' => object(SyliusComponentCoreContextShopperContext_000000007e497c57000000005871744d9ef656426ffaa21a64d67806c9aa9446), 'sonata_block' => object(GlobalVariables)))
        in [dev path]/sylius-standard/app/cache/dev/twig/de/de738443413a024e5f957d6221056a7447b6d257ab3f4594a97c5b5c475d16c6.php line 79

    at __TwigTemplate_c2fc9dfcccacf8f6ef07aa6ae60d1a3dbfadabedd5f6b3a92f6847d3a6d998d1->block_content(array('configuration' => object(RequestConfiguration), 'metadata' => object(Metadata), 'resource' => object(Product), 'product' => object(Product), 'app' => object(AppVariable), 'shopperContext' => object(SyliusComponentCoreContextShopperContext_000000007e497c57000000005871744d9ef656426ffaa21a64d67806c9aa9446), 'sonata_block' => object(GlobalVariables)), array('title' => array(object(__TwigTemplate_cc8615f88b3e48c5b73db2424d0c9bf6c6bba27b702bafbb56b6f99c5e98f85a), 'block_title'), 'metatags' => array(object(__TwigTemplate_cc8615f88b3e48c5b73db2424d0c9bf6c6bba27b702bafbb56b6f99c5e98f85a), 'block_metatags'), 'stylesheets' => array(object(__TwigTemplate_cc8615f88b3e48c5b73db2424d0c9bf6c6bba27b702bafbb56b6f99c5e98f85a), 'block_stylesheets'), 'content' => array(object(__TwigTemplate_c2fc9dfcccacf8f6ef07aa6ae60d1a3dbfadabedd5f6b3a92f6847d3a6d998d1), 'block_content'), 'footer' => array(object(__TwigTemplate_cc8615f88b3e48c5b73db2424d0c9bf6c6bba27b702bafbb56b6f99c5e98f85a), 'block_footer'), 'javascripts' => array(object(__TwigTemplate_cc8615f88b3e48c5b73db2424d0c9bf6c6bba27b702bafbb56b6f99c5e98f85a), 'block_javascripts')))
        in [dev path]/sylius-standard/vendor/twig/twig/lib/Twig/Template.php line 186

    at Twig_Template->displayBlock('content', array('configuration' => object(RequestConfiguration), 'metadata' => object(Metadata), 'resource' => object(Product), 'product' => object(Product), 'app' => object(AppVariable), 'shopperContext' => object(SyliusComponentCoreContextShopperContext_000000007e497c57000000005871744d9ef656426ffaa21a64d67806c9aa9446), 'sonata_block' => object(GlobalVariables)), array('title' => array(object(__TwigTemplate_cc8615f88b3e48c5b73db2424d0c9bf6c6bba27b702bafbb56b6f99c5e98f85a), 'block_title'), 'metatags' => array(object(__TwigTemplate_cc8615f88b3e48c5b73db2424d0c9bf6c6bba27b702bafbb56b6f99c5e98f85a), 'block_metatags'), 'stylesheets' => array(object(__TwigTemplate_cc8615f88b3e48c5b73db2424d0c9bf6c6bba27b702bafbb56b6f99c5e98f85a), 'block_stylesheets'), 'content' => array(object(__TwigTemplate_c2fc9dfcccacf8f6ef07aa6ae60d1a3dbfadabedd5f6b3a92f6847d3a6d998d1), 'block_content'), 'footer' => array(object(__TwigTemplate_cc8615f88b3e48c5b73db2424d0c9bf6c6bba27b702bafbb56b6f99c5e98f85a), 'block_footer'), 'javascripts' => array(object(__TwigTemplate_cc8615f88b3e48c5b73db2424d0c9bf6c6bba27b702bafbb56b6f99c5e98f85a), 'block_javascripts')))
        in [dev path]/sylius-standard/app/cache/dev/twig/29/291d71543a89d3003ac138731f10470de94dd66ea4481b0556f4ef2141d59012.php line 92

    at __TwigTemplate_cc8615f88b3e48c5b73db2424d0c9bf6c6bba27b702bafbb56b6f99c5e98f85a->doDisplay(array('configuration' => object(RequestConfiguration), 'metadata' => object(Metadata), 'resource' => object(Product), 'product' => object(Product), 'app' => object(AppVariable), 'shopperContext' => object(SyliusComponentCoreContextShopperContext_000000007e497c57000000005871744d9ef656426ffaa21a64d67806c9aa9446), 'sonata_block' => object(GlobalVariables)), array('title' => array(object(__TwigTemplate_cc8615f88b3e48c5b73db2424d0c9bf6c6bba27b702bafbb56b6f99c5e98f85a), 'block_title'), 'metatags' => array(object(__TwigTemplate_cc8615f88b3e48c5b73db2424d0c9bf6c6bba27b702bafbb56b6f99c5e98f85a), 'block_metatags'), 'stylesheets' => array(object(__TwigTemplate_cc8615f88b3e48c5b73db2424d0c9bf6c6bba27b702bafbb56b6f99c5e98f85a), 'block_stylesheets'), 'content' => array(object(__TwigTemplate_c2fc9dfcccacf8f6ef07aa6ae60d1a3dbfadabedd5f6b3a92f6847d3a6d998d1), 'block_content'), 'footer' => array(object(__TwigTemplate_cc8615f88b3e48c5b73db2424d0c9bf6c6bba27b702bafbb56b6f99c5e98f85a), 'block_footer'), 'javascripts' => array(object(__TwigTemplate_cc8615f88b3e48c5b73db2424d0c9bf6c6bba27b702bafbb56b6f99c5e98f85a), 'block_javascripts')))
        in [dev path]/sylius-standard/vendor/twig/twig/lib/Twig/Template.php line 379

    at Twig_Template->displayWithErrorHandling(array('configuration' => object(RequestConfiguration), 'metadata' => object(Metadata), 'resource' => object(Product), 'product' => object(Product), 'app' => object(AppVariable), 'shopperContext' => object(SyliusComponentCoreContextShopperContext_000000007e497c57000000005871744d9ef656426ffaa21a64d67806c9aa9446), 'sonata_block' => object(GlobalVariables)), array('title' => array(object(__TwigTemplate_cc8615f88b3e48c5b73db2424d0c9bf6c6bba27b702bafbb56b6f99c5e98f85a), 'block_title'), 'metatags' => array(object(__TwigTemplate_cc8615f88b3e48c5b73db2424d0c9bf6c6bba27b702bafbb56b6f99c5e98f85a), 'block_metatags'), 'stylesheets' => array(object(__TwigTemplate_cc8615f88b3e48c5b73db2424d0c9bf6c6bba27b702bafbb56b6f99c5e98f85a), 'block_stylesheets'), 'content' => array(object(__TwigTemplate_c2fc9dfcccacf8f6ef07aa6ae60d1a3dbfadabedd5f6b3a92f6847d3a6d998d1), 'block_content'), 'footer' => array(object(__TwigTemplate_cc8615f88b3e48c5b73db2424d0c9bf6c6bba27b702bafbb56b6f99c5e98f85a), 'block_footer'), 'javascripts' => array(object(__TwigTemplate_cc8615f88b3e48c5b73db2424d0c9bf6c6bba27b702bafbb56b6f99c5e98f85a), 'block_javascripts')))
        in [dev path]/sylius-standard/vendor/twig/twig/lib/Twig/Template.php line 347

    at Twig_Template->display(array('configuration' => object(RequestConfiguration), 'metadata' => object(Metadata), 'resource' => object(Product), 'product' => object(Product), 'app' => object(AppVariable), 'shopperContext' => object(SyliusComponentCoreContextShopperContext_000000007e497c57000000005871744d9ef656426ffaa21a64d67806c9aa9446), 'sonata_block' => object(GlobalVariables)), array('content' => array(object(__TwigTemplate_c2fc9dfcccacf8f6ef07aa6ae60d1a3dbfadabedd5f6b3a92f6847d3a6d998d1), 'block_content')))
        in [dev path]/sylius-standard/app/cache/dev/twig/de/de738443413a024e5f957d6221056a7447b6d257ab3f4594a97c5b5c475d16c6.php line 27

    at __TwigTemplate_c2fc9dfcccacf8f6ef07aa6ae60d1a3dbfadabedd5f6b3a92f6847d3a6d998d1->doDisplay(array('configuration' => object(RequestConfiguration), 'metadata' => object(Metadata), 'resource' => object(Product), 'product' => object(Product), 'app' => object(AppVariable), 'shopperContext' => object(SyliusComponentCoreContextShopperContext_000000007e497c57000000005871744d9ef656426ffaa21a64d67806c9aa9446), 'sonata_block' => object(GlobalVariables)), array('content' => array(object(__TwigTemplate_c2fc9dfcccacf8f6ef07aa6ae60d1a3dbfadabedd5f6b3a92f6847d3a6d998d1), 'block_content')))
        in [dev path]/sylius-standard/vendor/twig/twig/lib/Twig/Template.php line 379

    at Twig_Template->displayWithErrorHandling(array('configuration' => object(RequestConfiguration), 'metadata' => object(Metadata), 'resource' => object(Product), 'product' => object(Product), 'app' => object(AppVariable), 'shopperContext' => object(SyliusComponentCoreContextShopperContext_000000007e497c57000000005871744d9ef656426ffaa21a64d67806c9aa9446), 'sonata_block' => object(GlobalVariables)), array('content' => array(object(__TwigTemplate_c2fc9dfcccacf8f6ef07aa6ae60d1a3dbfadabedd5f6b3a92f6847d3a6d998d1), 'block_content')))
        in [dev path]/sylius-standard/vendor/twig/twig/lib/Twig/Template.php line 347

    at Twig_Template->display(array('configuration' => object(RequestConfiguration), 'metadata' => object(Metadata), 'resource' => object(Product), 'product' => object(Product)))
        in [dev path]/sylius-standard/vendor/twig/twig/lib/Twig/Template.php line 358

    at Twig_Template->render(array('configuration' => object(RequestConfiguration), 'metadata' => object(Metadata), 'resource' => object(Product), 'product' => object(Product)))
        in [dev path]/sylius-standard/vendor/symfony/symfony/src/Symfony/Bridge/Twig/TwigEngine.php line 50

    at Symfony\Bridge\Twig\TwigEngine->render('SyliusShopBundle:Product:show.html.twig', array('configuration' => object(RequestConfiguration), 'metadata' => object(Metadata), 'resource' => object(Product), 'product' => object(Product)))
        in [dev path]/sylius-standard/vendor/symfony/symfony/src/Symfony/Bundle/TwigBundle/TwigEngine.php line 72

    at Symfony\Bundle\TwigBundle\TwigEngine->render('SyliusShopBundle:Product:show.html.twig', array('configuration' => object(RequestConfiguration), 'metadata' => object(Metadata), 'resource' => object(Product), 'product' => object(Product)))
        in [dev path]/sylius-standard/vendor/friendsofsymfony/rest-bundle/FOS/RestBundle/View/ViewHandler.php line 419

    at FOS\RestBundle\View\ViewHandler->renderTemplate(object(View), 'html')
        in [dev path]/sylius-standard/vendor/friendsofsymfony/rest-bundle/FOS/RestBundle/View/ViewHandler.php line 494

    at FOS\RestBundle\View\ViewHandler->initResponse(object(View), 'html')
        in [dev path]/sylius-standard/vendor/friendsofsymfony/rest-bundle/FOS/RestBundle/View/ViewHandler.php line 473

    at FOS\RestBundle\View\ViewHandler->createResponse(object(View), object(Request), 'html')
        in [dev path]/sylius-standard/vendor/friendsofsymfony/rest-bundle/FOS/RestBundle/View/ViewHandler.php line 358

    at FOS\RestBundle\View\ViewHandler->handle(object(View))
        in [dev path]/sylius-standard/vendor/sylius/sylius/src/Sylius/Bundle/ResourceBundle/Controller/ViewHandler.php line 50

    at Sylius\Bundle\ResourceBundle\Controller\ViewHandler->handle(object(RequestConfiguration), object(View))
        in [dev path]/sylius-standard/vendor/sylius/sylius/src/Sylius/Bundle/ResourceBundle/Controller/ResourceController.php line 191

    at Sylius\Bundle\ResourceBundle\Controller\ResourceController->showAction(object(Request))
        in  line 

    at call_user_func_array(array(object(ResourceController), 'showAction'), array(object(Request)))
        in [dev path]/sylius-standard/app/bootstrap.php.cache line 3238

    at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), '1')
        in [dev path]/sylius-standard/app/bootstrap.php.cache line 3197

    at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), '1', true)
        in [dev path]/sylius-standard/app/bootstrap.php.cache line 3351

    at Symfony\Component\HttpKernel\DependencyInjection\ContainerAwareHttpKernel->handle(object(Request), '1', true)
        in [dev path]/sylius-standard/app/bootstrap.php.cache line 2536

    at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
        in [dev path]/sylius-standard/web/app_dev.php line 42

    at require('[dev path]/sylius-standard/web/app_dev.php')
        in [dev path]/sylius-standard/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Resources/config/router_dev.php line 40
```

**Comment by gdecorbiac** [comment]
I don't know whether it's related or not (I think so), but here are some more informations :

- Go to : _/admin/products/_
- Click on "Manage variations" of a configurable product -> _/admin/products/1/variants/_
- Create a new variation -> _/admin/products/1/variants/new_

it raises an _InvalidArgumentException_

Full stacktrace :

```
[1] Symfony\Component\Form\Exception\InvalidArgumentException: Could not load type "sylius_product_option_value_collection"
    at n/a
        in [dev path]/sylius-standard/vendor/symfony/symfony/src/Symfony/Component/Form/FormRegistry.php line 92

    at Symfony\Component\Form\FormRegistry->getType('sylius_product_option_value_collection')
        in [dev path]/sylius-standard/vendor/symfony/symfony/src/Symfony/Component/Form/FormFactory.php line 107

    at Symfony\Component\Form\FormFactory->createNamedBuilder('options', 'sylius_product_option_value_collection', object(ArrayCollection), array('options' => object(PersistentCollection), 'auto_initialize' => false, 'data' => object(ArrayCollection)))
        in [dev path]/sylius-standard/vendor/symfony/symfony/src/Symfony/Component/Form/FormFactory.php line 48

    at Symfony\Component\Form\FormFactory->createNamed('options', 'sylius_product_option_value_collection', object(ArrayCollection), array('options' => object(PersistentCollection), 'auto_initialize' => false))
        in [dev path]/sylius-standard/vendor/sylius/sylius/src/Sylius/Bundle/ProductBundle/Form/EventSubscriber/BuildProductVariantFormSubscriber.php line 68

    at Sylius\Bundle\ProductBundle\Form\EventSubscriber\BuildProductVariantFormSubscriber->preSetData(object(FormEvent), 'form.pre_set_data', object(EventDispatcher))
        in  line 

    at call_user_func(array(object(BuildProductVariantFormSubscriber), 'preSetData'), object(FormEvent), 'form.pre_set_data', object(EventDispatcher))
        in [dev path]/sylius-standard/vendor/symfony/symfony/src/Symfony/Component/EventDispatcher/EventDispatcher.php line 184

    at Symfony\Component\EventDispatcher\EventDispatcher->doDispatch(array(array(object(AddCodeFormSubscriber), 'preSetData'), array(object(BuildProductVariantFormSubscriber), 'preSetData'), array(object(BuildPriceableFormSubscriber), 'preSetData')), 'form.pre_set_data', object(FormEvent))
        in [dev path]/sylius-standard/vendor/symfony/symfony/src/Symfony/Component/EventDispatcher/EventDispatcher.php line 46

    at Symfony\Component\EventDispatcher\EventDispatcher->dispatch('form.pre_set_data', object(FormEvent))
        in [dev path]/sylius-standard/vendor/symfony/symfony/src/Symfony/Component/EventDispatcher/ImmutableEventDispatcher.php line 43

    at Symfony\Component\EventDispatcher\ImmutableEventDispatcher->dispatch('form.pre_set_data', object(FormEvent))
        in [dev path]/sylius-standard/vendor/symfony/symfony/src/Symfony/Component/Form/Form.php line 342

    at Symfony\Component\Form\Form->setData(object(ProductVariant))
        in [dev path]/sylius-standard/vendor/symfony/symfony/src/Symfony/Component/Form/Form.php line 478

    at Symfony\Component\Form\Form->initialize()
        in [dev path]/sylius-standard/vendor/symfony/symfony/src/Symfony/Component/Form/FormBuilder.php line 226

    at Symfony\Component\Form\FormBuilder->getForm()
        in [dev path]/sylius-standard/vendor/symfony/symfony/src/Symfony/Component/Form/FormFactory.php line 40

    at Symfony\Component\Form\FormFactory->create('sylius_product_variant', object(ProductVariant), array())
        in [dev path]/sylius-standard/vendor/sylius/sylius/src/Sylius/Bundle/ResourceBundle/Controller/ResourceFormFactory.php line 49

    at Sylius\Bundle\ResourceBundle\Controller\ResourceFormFactory->create(object(RequestConfiguration), object(ProductVariant))
        in [dev path]/sylius-standard/vendor/sylius/sylius/src/Sylius/Bundle/ResourceBundle/Controller/ResourceController.php line 236

    at Sylius\Bundle\ResourceBundle\Controller\ResourceController->createAction(object(Request))
        in  line 

    at call_user_func_array(array(object(ResourceController), 'createAction'), array(object(Request)))
        in [dev path]/sylius-standard/app/bootstrap.php.cache line 3238

    at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), '1')
        in [dev path]/sylius-standard/app/bootstrap.php.cache line 3197

    at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), '1', true)
        in [dev path]/sylius-standard/app/bootstrap.php.cache line 3351

    at Symfony\Component\HttpKernel\DependencyInjection\ContainerAwareHttpKernel->handle(object(Request), '1', true)
        in [dev path]/sylius-standard/app/bootstrap.php.cache line 2536

    at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
        in [dev path]/sylius-standard/web/app_dev.php line 42

    at require('[dev path]/sylius-standard/web/app_dev.php')
        in [dev path]/sylius-standard/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Resources/config/router_dev.php line 40
```

**Comment by pamil** [comment]
@gdecorbiac the last issue is connected with #6150, I've removed it because it didn't make the Behat test suite fail :) It needs a scenario and reverting these changes.

**Comment by gdecorbiac** [comment]
@pamil : how can I help you ?

**Comment by gdecorbiac** [comment]
@pamil 

this is  the same bug than [#6224](https://github.com/Sylius/Sylius/issues/6224)

the last issue is due to a service that does not exists : sylius_product_option_value_collection
the correct name seems to be : sylius_product_option_value_choice

the following changes are needed :

_Sylius\Bundle\ProductBundle\Form\EventSubscriber\BuildProductVariantFormSubscriber_

```
... line 65
// If the product has options, lets add this configuration field.
        if ($product->hasOptions()) {
            $form->add($this->factory->createNamed('optionValues', 'sylius_product_option_value_choice', $productVariant->getOptionValues(), [
                'options' => $product->getOptions(),
                'auto_initialize' => false,
            ]));
        }
...

```
_spec\Sylius\Bundle\ProductBundle\Form\EventSubscriber\BuildProductVariantFormSubscriberSpec_

```
... line 65
$factory->createNamed(
            'optionValues',
            'sylius_product_option_value_choice',
            [$optionValue],
            [
                'options' => [$options],
                'auto_initialize' => false,
            ]
        )->shouldBeCalled()->willReturn($optionsForm);
...
```
It skips the _InvalidArgumentException_
But I'm facing a new error (probably due to the Rename options to optionValues  PR) :

`Symfony\Component\OptionsResolver\Exception\UndefinedOptionsException: The option "options" does not exist. Defined options are: "action", "allow_extra_fields", "attr", "auto_initialize", "block_name", "by_reference", "cascade_validation", "choice_attr", "choice_label", "choice_list", "choice_loader", "choice_name", "choice_translation_domain", "choice_value", "choices", "choices_as_values", "compound", "constraints", "csrf_field_name", "csrf_message", "csrf_protection", "csrf_provider", "csrf_token_id", "csrf_token_manager", "data", "data_class", "disabled", "empty_data", "empty_value", "error_bubbling", "error_mapping", "expanded", "extra_fields_message", "group_by", "inherit_data", "intention", "invalid_message", "invalid_message_parameters", "label", "label_attr", "label_format", "mapped", "max_length", "method", "multiple", "option", "pattern", "placeholder", "post_max_size_message", "preferred_choices", "property_path", "read_only", "required", "translation_domain", "trim", "upload_max_size_message", "validation_groups", "virtual".`

I'm still digging...

**Comment by gdecorbiac** [comment]
so I digg further and the next issue is in _Sylius\Bundle\ProductBundle\Form\Type\ProductOptionValueChoiceType_

ther is a typo in _configureOptions()_ method where options value are defined as "option" instead of "options".
So it should be :

```
public function configureOptions(OptionsResolver $resolver)
    {
        $choiceList = function (Options $options) {
            return new ObjectChoiceList(
                $options['options']->getValues(),
                'value',
                [],
                null,
                'id',
                PropertyAccess::createPropertyAccessor()
            );
        };

        $resolver
            ->setDefaults([
                'choice_translation_domain' => false,
                'choice_list' => $choiceList,
            ])
            ->setRequired([
                'options',
            ])
            ->addAllowedTypes([
                'options' => ProductOptionInterface::class,
            ])
        ;
    }
```

But it's not over ! :(
It raises now a _InvalidOptionsException_ :

`The option "options" with value Doctrine\ORM\PersistentCollection is expected to be of type "Sylius\Component\Product\Model\ProductOptionInterface", but is of type "Doctrine\ORM\PersistentCollection".`

**Comment by gdecorbiac** [comment]
@pamil : Do you have any clue on how to fix it ? I can't go further due to my lack of knowledge of Symfony & Sylius

**Comment by gdecorbiac** [comment]
fixed by PR [#6264](https://github.com/Sylius/Sylius/pull/6264)
